primitives/session/src/lib.rs

```diff
@@ -47,6 +47,12 @@ sp_api::decl_runtime_apis! {
 		/// Returns the list of public raw public keys + key type.
 		fn decode_session_keys(encoded: Vec<u8>) -> Option<Vec<(Vec<u8>, KeyTypeId)>>;
 	}
+
+	/// Sesson boundary runtime api
+	pub trait SessionBoundaryApi<BlockNumber: codec::Codec> {
+		/// Returns the block number at which the current session began
+		fn get_session_boundary() -> BlockNumber;
+	}
```

Comment:
Why is that its own runtime api?

IMO it should just be added to the Beefy runtime api.